pandas/core/reshape/concat.py

```diff
@@ -315,7 +315,7 @@ def __init__(
                 "Only can inner (intersect) or outer (union) join the other axis"
             )
 
-        if isinstance(objs, dict):
+        if isinstance(objs, Mapping):
```

Comment:
can you use Mapping from collections.abc here.

Reply:
even better, use is_dict_like which is our canonical test for these things

Reply:
> even better, use is_dict_like which is our canonical test for these things

According to the Python compatibility statement, Python 3.6+ is required, which has the collections abcs. I'm quite agnostic here, but using the collections.abc would feel more future-proof without introducing restrictions.

Reply:
> can you use Mapping from collections.abc here.

They are actually the same class, but sure, makes sense.